File: packages/live-update/ios/Plugin/Classes/Results/GetChannelResult.swift
import Foundation
import Capacitor

@objc public class GetChannelResult: NSObject, Result {
    let channel: String?

    init(channel: String?) {
        self.channel = channel
    }

    public func toJSObject() -> AnyObject {
        var result = JSObject()
        result["channel"] = channel == nil ? NSNull() : channel
        return result as AnyObject
    }
}

File: packages/live-update/android/src/main/java/io/capawesome/capacitorjs/plugins/liveupdate/classes/results/GetChannelResult.java
package io.capawesome.capacitorjs.plugins.liveupdate.classes.results;

import androidx.annotation.Nullable;
import com.getcapacitor.JSObject;
import io.capawesome.capacitorjs.plugins.liveupdate.interfaces.Result;

public class GetChannelResult implements Result {

    @Nullable
    private String channel;

    public GetChannelResult(@Nullable String channel) {
        this.channel = channel;
    }

    public JSObject toJSObject() {
        JSObject result = new JSObject();
        result.put("channel", channel == null ? JSONObject.NULL : channel);
        return result;
    }
}
